src/client/activation/migrateDataScienceSettingsService.ts

```diff
@@ -26,16 +26,17 @@ export class MigrateDataScienceSettingsService implements IExtensionActivationSe
     ) {}
 
     public async activate(resource: Resource): Promise<void> {
-        this.updateSettings(resource).ignoreErrors();
+        await this.updateSettings(resource);
```

Comment:
I'm guessing we need this. But I'm pretty sure this does slow down activation. I take it that this was an issue as other bits were activating before settings were migrated? If it's only slow once that's probably ok.

Reply:
I agree. We might want to re-view the code, as we're looking for files in the method `updateSettings`.
Instead we should check if we have performed the check previously & short circuit (check some flag in global memento) if not alraedy doing that.

Reply:
IO on windows makes a big difference.